pyiron_base/interfaces/has_dict.py

```diff
@@ -25,10 +41,10 @@ def from_dict(self, obj_dict: dict, version: str = None):
         pass
 
     @abstractmethod
-    def to_dict(self):
+    def _to_dict(self):
         pass
 
-    def _type_to_dict(self):
+    def to_dict(self):
```

Comment:
```suggestion
    @HasDict.to_dict
    def to_dict(self):
```
What about this so people don't have to put `_...?`

Reply:
Do literature search what is common in the community?  Some reference to OOP best practices?

Reply:
I poked around for this, but I found nothing conclusive. The best I got was [this SO thread](https://stackoverflow.com/questions/30599316/using-abstract-private-methods-as-an-interface) from eight years ago; they lay out the same choices we're already aware of (OP's default was to do it the way you have here, if that's anything), but the question is left unanswered as "well, that's opinion". Personally, I find the `@abstractmethod` flag on the private method of the same name the clearest...but I agree it's opinion.

Reply:
I would like to stick to the current convention then.  Since this is a private class used internally anyway, it should be fairly easy to change in the future anyway, in case we decide differently.